metadata/metadata.go

```diff
@@ -22,6 +22,48 @@ import (
 	"strings"
 )
 
+// HasAnnotation checks if a given annotation exists
+func HasAnnotation(object v1.Object, key string) bool {
+	annotations, err := GetAnnotationsWithPrefix(object, key)
+	if err != nil || len(annotations) == 0 {
+		return false
+	}
+	return true
+}
+
+// HasAnnotationWithValue checks if an annotation exists by searching the key/ value
+func HasAnnotationWithValue(object v1.Object, key, value string) bool {
+	if annotations, err := GetAnnotationsWithPrefix(object, key); err == nil {
+		for _, val := range annotations {
+			if val == value {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// HasLabel checks if a given Label exists
+func HasLabel(object v1.Object, key string) bool {
+	labels, err := GetLabelsWithPrefix(object, key)
+	if err != nil || len(labels) == 0 {
+		return false
+	}
+	return true
+}
+
+// HasLabelWithValue checks if a Label exists by searching the key/ value
+func HasLabelWithValue(object v1.Object, key, value string) bool {
+	if labels, err := GetLabelsWithPrefix(object, key); err == nil {
+		for _, val := range labels {
+			if val == value {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
In a similar way to the annotation functions I'd do this:
```
labelValue, ok := obj.GetLabels()[key]
return ok && labelValue == value
```

Reply:
Updated. Thanks